src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -252,45 +253,67 @@ private async Task<string> GetLatestMatchingBundleVersionAsync(HttpClient httpCl
 
         internal string FindBestVersionMatch(VersionRange versionRange, IEnumerable<string> versions, string bundleId, FunctionsHostingConfigOptions configOption)
         {
+            int dotnetVersion = typeof(string).Assembly.GetName().Version.Major;
+
+            // Limit the version range for v4.x bundles on .NET 6 to be between 4.2.0 and 4.22.0
+            // Return original version range if not .NET 6 or not v4.x bundle
+            var effectiveVersionRange = GetAdjustedVersion(dotnetVersion, versionRange, bundleId);
+
+            // Check if there is a max version configured in hosting config for the current dotnet version and bundle major version
+            if (TryGetMaxBundleVersionFromHostingConfig(bundleId, versionRange.MinVersion.Major, dotnetVersion, configOption, out NuGetVersion maxBundleVersion))
+            {
+                effectiveVersionRange = new VersionRange(effectiveVersionRange.MinVersion, effectiveVersionRange.IsMinInclusive, maxBundleVersion, true);
+            }
+
             var bundleVersions = versions.Select(p =>
             {
                 var dirName = Path.GetFileName(p);
                 NuGetVersion.TryParse(dirName, out NuGetVersion version);
                 if (version != null)
                 {
-                    version = versionRange.Satisfies(version) ? version : null;
+                    version = effectiveVersionRange.Satisfies(version) ? version : null;
                 }
                 return version;
             }).Where(v => v != null).OrderByDescending(version => version.Version).ToList();
 
             var matchingVersion = ResolvePlatformReleaseChannelVersion(bundleVersions);
 
-            if (bundleId != ScriptConstants.DefaultExtensionBundleId)
-            {
-                return matchingVersion?.ToString();
-            }
+            return matchingVersion?.ToString();
+        }
 
-            // Check to see if there is a max bundle version set via hosting configuration, if yes then use that instead of the one
-            // available on VM or local machine. Only use MaximumBundleV3Version or MaximumBundleV4Version if the version configured
-            // by the customer resolved to version higher than the version set via hosting config.
-            if (!string.IsNullOrEmpty(configOption.MaximumBundleV3Version)
-                && matchingVersion?.Major == ScriptConstants.ExtensionBundleV3MajorVersion)
+        // Applies bundle version limits for .NET 6, using the default bundle if referencing a v4 major bundle version.
+        private VersionRange GetAdjustedVersion(int dotnetVersion, VersionRange versionRange, string bundleId)
+        {
+            if (!string.Equals(dotnetVersion, 6)
```

Comment:
Should have mentioned in the previous comment, but the key reason to use this here is to make sure you're controlling the string comparison. Please use the overload that takes the explicit `StringComparison` here.

The `bundleId` comparison should also use `string.Equals` (with the same overload mentioned above).